crud/common/types.lua

```diff
@@ -0,0 +1,109 @@
+local errors = require('errors')
+
+local collations = require('crud.common.collations')
+local TypeMismatchError = errors.new_class('TypeMismatchError')
+local UnsupportedCollationError = errors.new_class('UnsupportedCollationError')
+
+local types = {}
+
+local function comparing_nil(cmp)
```

Comment:
Is it work for all cmp functions or only for `lt`?

Reply:
it works for any, for example, `lt_boolean` and `lt_uuid` are decorated with it

Reply:
It's not "comparing_nil" it's "comparing_nullable".

So, your usage of this function is a bit awful. For each call you create new closure (at least it makes suffer Lua GC).
Seems you need to predefine comparators and then use them.